### DISCUSSION TITLE
[BUG] Get-PnPListItem error: "Unable to read data from the transport connection: An existing connection was forcibly closed by the remote host" if the DocLib has too many items.

### DISCUSSION THREAD

**Discussion by amaru96** [pnp/powershell#3656] [General]
If my doc lib has a massive amount of items (millions) the Get-PnPListItems command throws an error: Unable to read data from the transport connection: An existing connection was forcibly closed by the remote host.

This is the command I'm running which works most of the time, but as I said, if the doclib is massive, it errors.

Get-PnPListItem -List $doclib -Fields "FileLeafRef", "FileRef" -PageSize 100

**Comment by AndersRask** [comment]
Have you tried setting page size to 4000?
I haven't encountered a large list yet this did not work for, even if it
took a very long time.
Does the list have indexes?

tors. 28. dec. 2023 12.14 skrev amaru96 ***@***.***>:
